src/components/Filters/SingleSelectFilter.tsx

```diff
@@ -0,0 +1,43 @@
+import { Key } from "react";
+import { Filter } from "src/components/Filters/types";
+import { SelectField, SelectFieldProps } from "src/inputs/SelectField";
+import { Value } from "src/inputs/Value";
+import { defaultLabel } from "src/utils/defaultLabel";
+import { defaultTestId } from "src/utils/defaultTestId";
+import { TestIds } from "src/utils/useTestIds";
+
+export type SingleFilterProps<O, V extends Value> = Omit<SelectFieldProps<O, V>, "value" | "onSelect"> & {
+  defaultValue?: V;
+};
+
+export function singleFilter<O, V extends Key>(props: SingleFilterProps<O, V>): (key: string) => Filter<V> {
+  return (key) => new SingleFilter(key, props);
+}
+
+class SingleFilter<O, V extends Key> implements Filter<V> {
+  constructor(private key: string, private props: SingleFilterProps<O, V>) {}
+
+  render(value: V | undefined, setValue: (value: V | undefined) => void, tid: TestIds, inModal: boolean) {
+    const { label, defaultValue, ...props } = this.props;
+    return (
+      <SelectField<O, V>
+        {...props}
+        compact
+        value={value}
+        label={this.label}
+        inlineLabel
+        sizeToContent={!inModal}
+        onSelect={(value) => setValue(value || undefined)}
```

Comment:
Re: `value || undefined`, thinking about the case where a SelectField might have a boolean value, though I actually don't think that'd be possible, should use BooleanFilter in that case. Hmm... probably fine.

Reply:
Oh yeah...now that I know `??` works on `null` to, this should be `??`...will work in...